modules/tensor_mechanics/src/materials/ADComputeMultipleInelasticStress.C

```diff
@@ -140,10 +140,22 @@ ADComputeMultipleInelasticStress::computeQpStressIntermediateConfiguration()
 
     // If the elasticity tensor values have changed and the tensor is isotropic,
     // use the old strain to calculate the old stress
-    if (_is_elasticity_tensor_guaranteed_isotropic || !_perform_finite_strain_rotations)
-      _stress[_qp] = _elasticity_tensor[_qp] * (_elastic_strain_old[_qp] + _strain_increment[_qp]);
+    if (_use_old_elasticity_tensor)
+    {
+      if (_is_elasticity_tensor_guaranteed_isotropic || !_perform_finite_strain_rotations)
+        _stress[_qp] =
+            _elasticity_tensor_old[_qp] * (_elastic_strain_old[_qp] + _strain_increment[_qp]);
```

Comment:
You know, I thought making this a non AD rank four would help speed things up, but the timing is pretty similar...what am I missing @dschwen?

Reply:
Rank four * ADRankTwo two hurts...

Reply:
Your result is still going to be an AD object, so you still need to compute all derivative vector entries. That computation is going to be a tad simpler (one multiplication rather than two multiplications and one add), but the cost difference in these simple math functions is probably not that big compared to the cost of iterating over all the derivative vector entries and performing all that memory access (significant if your are not working with cached data!). Our code is not optimized for cache.